### DISCUSSION TITLE
Data updating on published page

### DISCUSSION THREAD

**Discussion by mouettxy** [Budibase/budibase#10981] [Help] [answered]
Hello,

I'm really frustrated with data updating in published page, I created a table, and after updating data source (i'm using internal DB), nothing happens on published page after reload :(

There's really no way to do it?

**Comment by ConorWebb96** [comment]
Hey @mouettxy,

I'm not really sure what you are referring to. Could you provide some additional detail, possibly some screenshots or even a video to go along with your question?

Thanks ahead of time.

**Comment by mouettxy** [comment]
@ConorWebb96 Thanks for blazing fast answer! There's screencast of problem that I'm trying to describe. Notice that after adding rows to table, published page not reflecting the changes

https://github.com/Budibase/budibase/assets/32680159/104c97eb-864c-40b5-bace-5a171b1289ff

**Reply by ConorWebb96** [reply, marked as answer]
Hey @mouettxy,

I understand now thank you for the video!

In Budibase, there are two different databases: one for development and one for production.

The development database is where you make changes and see them right away. You can build and test everything here. Once you're happy with everything, you just need to click "publish" to sync all the data and changes to your production app. Remember, if you don't click "publish" in the development database, your changes won't be synced.

In the production database, all the data is automatically synced back to the development database. This means that any new rows created, edited, or deleted through forms will show up in the development database as soon as they are made.

I understand that this may seem confusing, but I want to assure you that we have plans to improve the process in the future, hopefully making it more intuitive.

I hope this explanation makes sense. If you have any more questions, please let us know.

**Reply by mouettxy** [reply]
@ConorWebb96 Many thanks for clear explanation. So to summarize this up, if I add rows to table via Public API (on my site, there's form that tied to budibase table), for changes to show up on Published page, I need to re-publish manually?

**Reply by mouettxy** [reply]
@ConorWebb96 Oh, your reply give me some idea, before I have an App ID that looks like this app_dev_*_*, if I remove _dev part, changes sucessefully went to published page without any problems.

Thank you so much, wonder if this discussion would be helpful to others!

**Reply by ConorWebb96** [reply]
Hey @mouettxy,

Was just about to let you know that haha. Happy you got it sorted! 👍
